pkg/controller/config/config.go

```diff
@@ -0,0 +1,236 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package config
+
+import (
+	"fmt"
+	"net"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-network-operator/pkg/render"
+	"gopkg.in/yaml.v2"
+
+	operatorv1 "github.com/ruicao93/antrea-operator/pkg/apis/operator/v1"
+	"github.com/ruicao93/antrea-operator/pkg/types"
+)
+
+func FillDefaults(clusterConfig *configv1.Network, operConfig *operatorv1.AntreaInstall) error {
+	antreaAgentConfig := make(map[string]interface{})
+	err := yaml.Unmarshal([]byte(operConfig.Spec.AntreaAgentConfig), &antreaAgentConfig)
+	if err != nil {
+		return fmt.Errorf("failed to parse antrea agent-config: %v", err)
+	}
+
+	// Set service CIDR.
+	if len(clusterConfig.Spec.ServiceNetwork) == 0 {
+		return fmt.Errorf("service network can not be empty")
+	}
+	serviceCIDR, ok := antreaAgentConfig[types.ServiceCIDROption]
+	if ok {
+		found := false
+		for _, serviceNet := range clusterConfig.Spec.ServiceNetwork {
+			if serviceNet == serviceCIDR {
+				found = true
+				break
+			}
+		}
+		if !found {
+			log.Info("WARNING: option: %s is overwritten by cluster config")
+			antreaAgentConfig[types.ServiceCIDROption] = clusterConfig.Spec.ServiceNetwork[0]
+		}
+	} else {
+		antreaAgentConfig[types.ServiceCIDROption] = clusterConfig.Spec.ServiceNetwork[0]
+	}
+
+	// Set default MTU.
+	_, ok = antreaAgentConfig[types.DefaultMTUOption]
+	if !ok {
+		antreaAgentConfig[types.DefaultMTUOption] = types.DefaultMTU
+	}
+
+	// Set Antrea image.
+	if operConfig.Spec.AntreaImage == "" {
+		operConfig.Spec.AntreaImage = types.DefaultAntreaImage
+	}
+	if operConfig.Spec.AntreaImage != types.DefaultAntreaImage {
+		log.Info("WARNING: expected Antrea Image is: %s, but got: %s", types.DefaultAntreaImage, operConfig.Spec.AntreaImage)
+	}
+
+	updatedAntreaAgentConfig, err := yaml.Marshal(antreaAgentConfig)
+	if err != nil {
+		return fmt.Errorf("failed to fill default serviceCIDR in antrea-agent config: %v", antreaAgentConfig)
+	}
+	operConfig.Spec.AntreaAgentConfig = string(updatedAntreaAgentConfig)
+	return nil
+}
+
+func ValidateConfig(clusterConfig *configv1.Network, operConfig *operatorv1.AntreaInstall) error {
+	var errs []error
+	// Validate cluster network config
+	if clusterConfig.Spec.NetworkType != types.NetworkType {
+		errs = append(errs, fmt.Errorf("expected network type is: %s, but got: %s", types.NetworkType, clusterConfig.Spec.NetworkType))
+	}
+	if len(clusterConfig.Spec.ServiceNetwork) == 0 {
```

Comment:
FillDefaults also has this check

Reply:
Sure, I will remove the network CRD config check in `ValidateConfig()`.